### PR TITLE
generalize interpolation search

### PR DESCRIPTION
**Pull request by iattempt**
**Fixes issue:**
**Changes:**
1. update to template version

### CODE REVIEW

File: code/search/interpolation_search/interpolation_search.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+template<typename _Random_Access_Iter,
+         typename _Type = typename std::iterator_traits<_Random_Access_Iter>::value_type,
+         typename _Compare>
+_Random_Access_Iter
+interpolationSearch(_Random_Access_Iter begin,
+                     _Random_Access_Iter end,
+                     _Type const &find,
+                     _Compare compare)
+{
+    auto not_found = end;
```
Comment (arnavb) [inline code/search/interpolation_search/interpolation_search.cpp]: You aren't following the style guide. Use camelCase.

File: code/search/interpolation_search/interpolation_search.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        // TODO: replace '<=' with '!=' or else to be more useful,
+        // (e.g., allow input iterator can be passed)
+        while (begin <= end)
+        {
+            // Now we will enquire the position keeping in mind the uniform distribution in mind
+            auto pos = begin;
+            if (*end - *begin)
+            {
+                auto len = (ptrdiff_t)std::distance(begin, end) * (find - *begin) / (*end - *begin);
```
Comment (arnavb) [inline code/search/interpolation_search/interpolation_search.cpp]: Use static_cast for C++

File: code/search/interpolation_search/interpolation_search.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        --end;
+
+        // TODO: replace '<=' with '!=' or else to be more useful,
+        // (e.g., allow input iterator can be passed)
+        while (begin <= end)
+        {
+            // Now we will enquire the position keeping in mind the uniform distribution in mind
+            auto pos = begin;
+            if (*end - *begin)
```
Comment (arnavb) [inline code/search/interpolation_search/interpolation_search.cpp]: By explicit about what you are checking here. (Whether *end - *begin >= 1)

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**